src/util/sharing_node.h

```diff
@@ -356,8 +356,20 @@ SN_TYPE_PAR_DECL class d_leaft : public small_shared_pointeet<unsigned>
 {
 public:
 #if SN_SHARE_KEYS == 1
+  d_leaft(const std::shared_ptr<keyT> &k, const valueT &v) : k(k), v(v)
+  {
+  }
+  d_leaft(const std::shared_ptr<keyT> &k, valueT &&v) : k(k), v(std::move(v))
+  {
+  }
   std::shared_ptr<keyT> k;
 #else
+  d_leaft(const keyT &k, const valueT &v) : k(k), v(v)
+  {
+  }
+  d_leaft(const keyT &k, valueT &&v) : k(k), v(std::move(v))
```

Comment:
Instead of duplicating all the declarations, couldn't we just a single `typedef` (or `using`) that holds the effective key type? After all `k` is a public member as well, and has that effective key type, so it would be beneficial to just consistently use that typedef name.

Reply:
Done